### DISCUSSION TITLE
Jamulus 3.7.0 – Error Upgrading

### DISCUSSION THREAD

**Discussion by PatrickP-us** [jamulussoftware/jamulus#1309] [Technical Support & Help] [answered]
Hello,

I've used Simon Tomlinson's easy to follow instructions a few times previously to successful upgrade my Jamulus server EC2 Ubuntu instance . However, when moving to 3.7.0 it is stopping at the following error:

     /usr/lib/qt5/bin/lrelease src/res/translation/translation_de_DE.ts -qm src/res/translation/translation_de_DE.qm
     make: /usr/lib/qt5/bin/lrelease: Command not found
     make: *** [Makefile:1316: src/res/translation/translation_de_DE.qm] Error 127
     mv: cannot move 'llcon-jamulus/' to '/usr/local/bin/llcon-jamulus.3.7.0/llcon-jamulus': Directory not empty

Any guidance in resolving the issue would be greatly appreciated. Also, thanks to all that have made Jamulus available to the world...it is remarkable!!

Patrick

**Comment by ann0see** [comment]
I think that's a bug. See the issues tab on this repository. There might already be an open issue: https://github.com/jamulussoftware/jamulus/issues/1304

You don't need to compile jamulus anymore from now on. 

You can just remove the binary/setup files/... and follow the Linux/Debian instructions from jamulus.io to install the .deb file.

**Reply by PatrickP-us** [reply]
Hello,

First, thank you for the quick reply. I have a follow up as I'm pretty familiar with macOS, but not Ubuntu Linux. Can you point me to instructions as to how to remove the specific binary/setup files/... ?

Again, your guidance is much appreciated!

Patrick

**Reply by ann0see** [reply]
Hi Patrick,

I'm not 100% sure how to remove the old install, but according to the file Jamulus.pro in this repository, you should be able to remove the Jamulus binary in /usr/local/bin/Jamulus

And just follow https://jamulus.io/wiki/Server-Linux

**Comment by softins** [comment]
As an alternative to @ann0see's reply, you can get the latest version of Simon's script, which I updated yesterday to fix a couple of issues and to be compatible with the latest release. It is at https://worldjam.vip/jamulus/setup-jamulus.sh

On Ubuntu 20, an additional package needs to be installed, `qttools5-dev-tools`, to provide the `lrelease` tool. The updated script takes care of that, and has some other corrections too.

**Reply by PatrickP-us** [reply]
Hello,

Thanks much for the additional option! I was able to delete the 3.7.0 directory and I thought it would then revert back to 3.6.2, but no go (as you can see that and the previous 3.6.1 one also in the same place). My Mac terminal is further having issues reading the "jamulus_3.7.0_ubuntu_amd64.deb" file. I know I'm giving it the correct path, but it no matter what I do it says it is an "unsupported file".

Hopefully Simon's new script will get me on the mend!

Patrick

**Reply by ann0see** [reply]
Not exactly sure, what you were trying. but .deb files are for Debian/Ubuntu not macOS.

**Reply by PatrickP-us** [reply]
Tony & @ann0see,

I'm up and running and your support in making that happen is greatly appreciated!!

I guess as a by product I now now where the old Jamulus installations are stored, so if I need to free up space on the AWS EC2 Instance I can do so. I take it the ability to delete a version (and revert) is what Simon is referencing when he stated: "Note this script does not remove your existing Jamulus server, it creates a new version so you can easily go back to an old version if you want."

Two things I noted after rebooting the EC2 Instance a coupe of time are:
     •Memory usage appears be be a bit higher (Memory usage: 45% vs. 28% to 35% previously)
     •Ping time and delay look to be a little better (could be just less traffic today, but might be due to the update)

Have a great weekend!

Patrick

**Reply by PatrickP-us** [reply]
I was just trying to execute the "sudo apt install /path/to/jamulus_3.7.0_ubuntu_amd64.deb" command. I know I was giving the correct file path. As I recall it was tricky setting up the PEM key on the Mac for the Ubuntu EC2 Instance as well (that file is in the same location). Perhaps a .deb simply cannot be read from a Mac (even by Terminal).

Thanks again!

Patrick

**Reply by ann0see** [reply]
Great! Could you please flag the answer which solved your issue or give a brief comment about what you did and flag this as answer? This allows others to find solutions a lot quicker.

**Comment by PatrickP-us** [comment, marked as answer]
Certainly. The approach I used was logging into the EC2 Instance then using the updated version of Simon's script that Tony provided. In addition, the rest of the command language to trigger the script that Simon previously posted on Facebook. Here it is:

wget https://worldjam.vip/jamulus/setup-jamulus.sh
chmod 744 setup-jamulus.sh
./setup-jamulus.sh

In macOS that of course avoids the issues with trying to access Debian/Ubuntu files locally. That said, as long as the script continues to be kingly updated with new releases, I would think that it would be the easiest way to do an install on Windows or Linux too.

Take care,

Patrick

**Reply by ann0see** [reply]
Great! Thank you for this explanation.
